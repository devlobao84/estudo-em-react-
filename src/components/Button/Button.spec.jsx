<<<<<<< HEAD
 
=======
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from ".";

describe("<button />", () => {
  it('should render the button with the text "Load More', () => {
    render(<Button text="Load More" />);
    expect.assertions(1);
    const button = screen.getByRole("button", { name: /load more/i });
    expect(button).toBeInTheDocument();
  });

  it("should call function on button click", () => {
    const fn = jest.fn();
    render(<Button text="Load More" onClick={fn} />);
    const button = screen.getByRole("button", { name: /load more/i });
    userEvent.click(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });

  //it("shound be disabled when disabled is true", () => {    
  //  render(<Button text="Load more" disabled={true} />);
  //  const button = screen.getByRole("button", { name: /load more/i });
  //  expect(button).toBeEnabled();
  //});

  it("shound be enabled when disabled is false", () => {
    const fn = jest.fn();
    render(<Button text="Load more" disabled={false} onClick={fn} />);
    const button = screen.getByRole("button", { name: /load more/i });
    expect(button).toBeEnabled();
  });

  it("shound match snapshot", () => {
    const fn = jest.fn();
    const {container} = render(<Button text="Load more" disabled={false} onClick={fn} />);
    expect(container.firstChild).toBeEnabled();
  });

});

>>>>>>> 690d07f0ecaacf31bd3425b6caf92e5410511ec2
