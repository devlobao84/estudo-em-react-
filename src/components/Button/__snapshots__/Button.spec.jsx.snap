// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<button /> shound match snapshot 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
